### DISCUSSION TITLE
How to create PolygonInfo with points list?

### DISCUSSION THREAD

**Discussion by antn9x** [axmolengine/axmol#1298] [General]
Now we can get PolygonInfo with `AutoPolygon::generatePolygon`. But how to create `PolygonInfo ` with `vector<Vec2>`?

**Comment by antn9x** [comment]
And how can we calculate `triangles`, `vertices`, `verticesUV` in plist file with points list?

**Comment by aismann** [comment]
@antn9x 
Have you look on the SpritePolygohn on cpp-tests?
You have the implement some "own functions" which return what you want I think.

**Reply by antn9x** [reply]
Yes, I read cpp-tests and `AutoPolygon` source code, but hard to convert to js. I target to write my own tool in js.

**Reply by aismann** [reply]
sorry here is more `cpp `the native programing language. Maybe another one can help you?

**Reply by antn9x** [reply]
ok, nodejs can call cpp code too. How can I extract AutoPolygon logic to stand alone? Can you help?

**Reply by aismann** [reply]
One of some more possible implementatioins:
You can make an own "extension" of the "`AutoPolygon`", "`PolygonInfo`" as basis classes or create copies.
Add your own methods (`getTriangles`, `getVertices`, `getVerticesUV`) and call these from nodejs.
Thats all. 

Another way is to implement this "methods" on the original classes and make a PR
